algorithms/sorting/pdqsort.py

```diff
@@ -0,0 +1,177 @@
+import random
+import unittest
+
+
+def insertion_sort(arr, left, right):
```

Comment:
The function prototype should be like this, note that I also add type hints to make the code more clear.
```suggestion
def insertion_sort(arr: list[T], left: int, right: int) -> None:
```